src/key/key.js

```diff
@@ -268,6 +264,9 @@ class Key {
 
     try {
       const primaryUser = await this.getPrimaryUser(date, userID, config);
+      if (date && primaryKey.getCreationTime() > date) {
+        throw new Error('Primary key was created after the given date');
+      }
```

Comment:
This seems very ad-hoc, and should be checked somewhere else, most likely in `verifyPrimaryKey` - in fact, I think it will currently already throw "Primary key is expired" in this situation?